.github/workflows/yamllint.yaml

```diff
@@ -0,0 +1,65 @@
+name: yamllint
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    types: [ opened, edited, reopened, synchronize, ready_for_review ]
+
+jobs:
+  yamllint-check:
+    name: yaml-lint
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install yamllint
+      run: sudo apt-get update && sudo apt-get install -y yamllint
+
+    - uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4.3.0
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Extract YAML snippets from md files
+      run: |
+        set -euo pipefail
+        mkdir -p .yaml-snippet
+
+        # Process each md file, excluding docs/user-guide/src/quick-start.md
+        while IFS= read -r md_file; do
+          echo "Processing: ${md_file}"
+
+          # Get safe filename based on full path
+          safe_name=$(echo "${md_file}" | sed 's/[\/\.]/_/g')
+
+          # Extract YAML code blocks and save each one
+          awk -v base="${safe_name}" '
+            BEGIN { count=0 }
+            /^```(yaml|yml)/ { in_block=1; next }
+            /^```/ && in_block {
+              if (block != "") {
+                count++
+                file = sprintf(".yaml-snippet/%s_snippet_%03d.yaml", base, count)
+                print block > file
+                close(file)
+                block = ""
+              }
+              in_block=0
+              next
+            }
+            in_block { block = (block == "" ? $0 : block "\n" $0) }
+          ' "${md_file}"
+        done < <(find . -name "*.md" -not -path "./.git/*" -not -path "./docs/user-guide/src/quick-start.md")
```

Comment:
This part is needed to exclude the syntax in `/docs/user-guide/src/quick-start.md` file.  If we reuse the workflow, it was not possible to exclude this file check during snipped yaml code collection..